src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,320 @@
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+    return summary_df
+
+
+def write_results(
+    result: EvaluationResult, file_formats: List[str], output_dir: str, model_name: str
+) -> List[str]:
+    validate_output_file_formats(file_formats)
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    now = datetime.now()
+    timestamp = now.isoformat()
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    results_files = []
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.JSONL.value, output_dir, timestamp, model_name
+            )
+            response_df.to_json(f"{results_file}", orient="records", lines=True)
+            results_files.append(results_file)
+        elif IOFileType.CSV.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.CSV.value, output_dir, timestamp, model_name
+            )
+            response_df.to_csv(f"{results_file}", index=False)
+            results_files.append(results_file)
+        elif IOFileType.XLSX.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.XLSX.value, output_dir, timestamp, model_name
+            )
+            create_excel_results_file(results_file, result)
+            results_files.append(results_file)
+
+    return results_files
+
+
+def run_local_model_evaluation(
+    ctx,
+    evaluator,
+    model,
+    model_name: str | None,
+    temperature,
+    system_prompt,
+    max_workers,
+    gpus,
+    backend,
+    enable_serving_output,
+    input_questions,
+    judge_model_name,
+    judge_openai_api_key,
+):
+    server = None
+    try:
+        logger.debug("Model being evaluated in DK-Bench is local")
+        if model_name is None:
+            model_name = get_local_model_name(model)
+        server, api_base, _ = launch_server(
+            eval_serve=ctx.obj.config.serve,
+            tls_client_cert=ctx.params["tls_client_cert"],
```

Comment:
@danmcp do you know why all of the other evaluations are using `ctx.params` as input to launch server (ex: https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/evaluate.py#L101) instead of the variables passed into `evaluate_data()` from the flags (ex: https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/evaluate.py#L51)?

I was just copying what the other benchmark implementations were doing with this.

Reply:
@alinaryan I see your work undid what @cdoern did here https://github.com/instructlab/instructlab/commit/9049733ebaa838047af4aa87a044d6b16ecdd0dd#diff-1bb428657457ab8d7d8393aec32de4e1510d03097af17d25445b97294e689255. Maybe it was just a mistake?